src/main/java/data/Recipe.java

```diff
@@ -0,0 +1,87 @@
+import java.util.LinkedList;
+import java.util.List;
+
+public class Recipe {
+
+  protected String name, description;
+  protected List<Step> steps;
+  private List<SpinOff> spinOffs;
+
+  public Recipe() {}
+  
+  public Recipe(String recipeName, String recipeDescription, List<Step> recipeSteps) {
+    this.name = recipeName;
+    this.description = recipeDescription;
+    this.steps = recipeSteps;
+    this.spinOffs = new LinkedList<>();
+  }
+
+  public void appendStep(Step newStep) {
```

Comment:
I think appendStep should be kept separate because it invokes add(newStep) on the LinkedList, which runs in O(1) time. Combining it with addStep would mean invoking add(steps.size(), newStep) on the LinkedList, which runs in O(n) time. Also, I designed addStep to automatically append the new step when the input position is greater than the size of the LinkedList because allowing the user to remove/reorder steps may cause discrepancies with the list size. If the input position is greater than the list size by mistake, the expected behavior is still to append the step to the list.